spinnaker/spinnaker_system/google_http_lb_test.py

```diff
@@ -0,0 +1,1256 @@
+
+    Args:
+      parser: argparse.ArgumentParser
+    '''
+    super(GoogleHttpLoadBalancerTestScenario, cls).initArgumentParser(
+      parser, defaults=defaults)
+
+
+  def __init__(self, bindings, agent=None):
+    '''Constructor.
+
+    Args:
+      bindings: [dict] The data bindings to use to configure the scenario.
+      agent: [GateAgent] The agent for invoking the test operations on Gate.
+    '''
+    super(GoogleHttpLoadBalancerTestScenario, self).__init__(bindings, agent)
+
+    bindings = self.bindings
+
+    self.__lb_detail = 'lb'
+    self.TEST_APP = bindings['TEST_APP']
+    self.__lb_name = '{app}-{stack}-{detail}'.format(
+        app=bindings['TEST_APP'], stack=bindings['TEST_STACK'],
+        detail=self.__lb_detail)
+    self.__default_hc = {
+      'name': 'basic-http-check',
+      'requestPath': '/',
+      'port': 80,
+      'checkIntervalSec': 1,
+      'timeoutSec': 1,
+      'healthyThreshold': 1,
+      'unhealthyThreshold': 1
+    }
+    self.__default_upsert = [
+      {
+        'cloudProvider': 'gce',
+        'provider': 'gce',
+        'stack': bindings['TEST_STACK'],
+        'credentials': bindings['GCE_CREDENTIALS'],
+        'region': bindings['TEST_GCE_REGION'],
+        'loadBalancerType': 'HTTP',
+        'loadBalancerName': self.__lb_name,
+        'portRange': '80',
+        'defaultService': {
+          'name': 'default-backend-service',
+          'backends': [],
+          'healthCheck': self.__default_hc,
+        },
+        'certificate': '',
+        'hostRules': [
+          {
+            'hostPatterns': ['host1.com', 'host2.com'],
+            'pathMatcher': {
+              'pathRules': [
+                {
+                  'paths': ['/path', '/path2/more'],
+                  'backendService': {
+                    'name': 'backend-service',
+                    'backends': [],
+                    'healthCheck': self.__default_hc,
+                  }
+                }
+              ],
+              'defaultService': {
+                'name': 'pm-backend-service',
+                'backends': [],
+                'healthCheck': self.__default_hc,
+              }
+            }
+          }
+        ],
+        'type': 'upsertLoadBalancer',
+        'availabilityZones': {bindings['TEST_GCE_REGION']: []},
+        'user': '[anonymous]'
+      }
+    ]
+    self.__default_delete = [
+      {
+        'type': 'deleteLoadBalancer',
+        'cloudProvider': 'gce',
+        'loadBalancerType': 'HTTP',
+        'loadBalancerName': self.__lb_name,
+        'region': bindings['TEST_GCE_REGION'],
+        'regions': [bindings['TEST_GCE_REGION']],
+        'credentials': bindings['GCE_CREDENTIALS'],
+        'user': '[anonymous]'
+      }
+    ]
+
+
+  def _make_bs_link(self, bs):
+    '''Make a fully-formatted backend service link.
+    '''
+    return (GCE_URL_PREFIX + self.bindings['GCE_PROJECT'] +
+            '/global/backendServices/' + bs)
+
```

Comment:
I think these two methods (along with GCE_URL_PREFIX) belong in gcp_testing, probably gcp_compute_agent since these pertain to the Compute service and that agent is where knowledge of Compute lives.

I'm not sure offhand what the API would look like but suspect you would do a discovery document lookup as I do for the methods and resolve the variables into the URL. So assuming the discovery document tells you how to get the resource link (which I think is just the GET URL), this would look like:
agent.getResourceUrl(project=self.bindings['GCE_PROJECT'], resource='backendServices', resource_id=bs)

The knowledge of the actual URL comes from the discovery document, which comes directly from GCP at runtime and the variables are bound in by the caller.